index.js

```diff
@@ -26,9 +26,63 @@ const app = express()
 const port = PORT || 3000
 const queryLimit = parseInt(QUERY_DEPTH_LIMIT, 10) || 10 // Default to ten.
 
-if (NODE_ENV === "production") {
-  app.set("forceSSLOptions", { trustXFPHeader: true }).use(forceSSL)
-  app.set("trust proxy", 1)
+function startApp(schema) {
+  if (NODE_ENV === "production") {
+    app.set("forceSSLOptions", { trustXFPHeader: true }).use(forceSSL)
+    app.set("trust proxy", 1)
+  }
+
+  app.get("/favicon.ico", (_req, res) => {
+    res
+      .status(200)
+      .set({ "Content-Type": "image/x-icon" })
+      .end()
+  })
+
+  app.all("/graphql", (req, res) => res.redirect("/"))
+
+  app.use(bodyParser.json())
+  app.use(
+    "/",
+    cors(),
+    morgan,
+    graphqlHTTP((req, res) => {
+      info("----------")
+
+      legacyLoaders.clearAll()
+
+      const accessToken = req.headers["x-access-token"]
+      const userID = req.headers["x-user-id"]
+      const timezone = req.headers["x-timezone"]
+      const requestID = req.headers["x-request-id"] || "implement-me"
+
+      // Accepts a tz database timezone string. See http://www.iana.org/time-zones,
+      // https://en.wikipedia.org/wiki/List_of_tz_database_time_zones
+      let defaultTimezone
+      if (moment.tz.zone(timezone)) {
+        defaultTimezone = timezone
+      }
+
+      const loaders = createLoaders(accessToken, userID, requestID)
+      // Share with e.g. the Convection ApolloLink in mergedSchema.
+      res.locals.dataLoaders = loaders // eslint-disable-line no-param-reassign
+
+      return {
+        schema,
+        graphiql: true,
+        rootValue: {
+          accessToken,
+          userID,
+          defaultTimezone,
+          ...loaders,
+        },
+        formatError: graphqlErrorHandler(req.body),
+        validationRules: [depthLimit(queryLimit)],
```

Comment:
Nice! Wasn't aware of [GraphQL depth limit](https://github.com/stems/graphql-depth-limit).

Reply:
I… dislike it so much 😞 It’s such an arbitrary 🔨 

Of course I understand why we’d want it at the moment, but in the long term I see only allowing persisted queries in production (or limiting depth query for not persisted queries) as a better solution.